### PR TITLE
Kafka acl

### PR DESCRIPTION
**Pull request by h3nd24**

### CODE REVIEW

**Review by Mitchel-Stig-Instaclustr — CHANGES_REQUESTED** [review]

File: instaclustr/unit_test_utils.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package instaclustr
```
Comment (arjunrajinstaclustr) [inline instaclustr/unit_test_utils.go:1]: Since this is all Kafka specific, could we rename the file to kakfa_unit_test_utils?

**Review by arjunrajinstaclustr — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -3,8 +3,9 @@ BIN_NAME=terraform-provider-instaclustr
 
 
 # for VERSION, don't add prefix "v", e.g., use "1.9.8" instead of "v1.9.8" as it could break circleCI stuff
-VERSION=1.14.3
-INSTALL_FOLDER=$(HOME)/.terraform.d/plugins/terraform.instaclustr.com/instaclustr/instaclustr/$(VERSION)/darwin_amd64
+VERSION=1.14.6
+INSTALL_FOLDER=$(HOME)/.terraform.d/plugins/darwin_amd64
```
[the remark this answers is not in the document]
Comment (h3nd24) [reply Makefile:7]: yep

File: acc_test/data/kafka_acl_create_acl_duplicate.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
+// This is part of testing "kafka acl" suite, 2 of 3 
+provider "instaclustr" {
```
[the remark this answers is not in the document]
Comment (h3nd24) [reply acc_test/data/kafka_acl_create_acl_duplicate.tf:2]: This should have been combined with the existing Kafka resource test (currently user, but also with topic). If it's somehow different, I'll make the adjustment

File: docs/data-sources/kafka_acl.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+Kafka ACL list is a read-only data source used to get the list of kafka ACL in a cluster, 
+
+## Properties
+
+`instaclustr_kafka_acl_list`
+
+Property | Description | Default
+---------|-------------|--------
+`cluster_id`|The ID of an existing Instaclustr Kafka managed cluster. |Required
+
```
[the remark this answers is not in the document]
Comment (h3nd24) [reply docs/data-sources/kafka_acl.md:19]: ok

File: docs/resources/kafka_acl.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
+---
+page_title: "instaclustr_kafka_acl Resource - terraform-provider-instaclustr"
+subcategory: ""
+description: |-
+  
+---
+
+# Resource  `instaclustr_kafka_acl`
+Resources for managing Kafka ACL for a Kafka cluster. 
+Kafka ACL is a resource used to create and delete Kafka ACL.
```
[the remark this answers is not in the document]
Comment (h3nd24) [reply docs/resources/kafka_acl.md:10]: sounds reasonable, I'll try to merge it

File: instaclustr/api_client_kafka_unit_test.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package instaclustr
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+)
+
+func TestAPIDeleteKafkaAcl(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (h3nd24) [reply instaclustr/api_client_kafka_unit_test.go:9]: OK

File: instaclustr/unit_test_utils.go (hunk 1 above)
Comment (h3nd24) [reply instaclustr/unit_test_utils.go:1]: OK

**Review by arjunrajinstaclustr — APPROVED** [review]

**Review by YuWang24 — APPROVED** [review]

File: instaclustr/api_client_kafka_unit_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (h3nd24) [reply instaclustr/api_client_kafka_unit_test.go]: Sorry for whatever reason I thought I have resolved this but it turns out to be wrong. Resolving this for now.

**Review by Mitchel-Stig-Instaclustr — APPROVED** [review]

**Pull request merged**